tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -255,24 +283,13 @@ public async Task<long> UploadRepositoryChangesAsync()
             Log.Error(ex, "Error detecting and reconfiguring git repository for shallow clone.");
         }
 
-        var gitLogOutput = await RunGitCommandAsync("log --format=%H -n 1000 --since=\"1 month ago\"", MetricTags.CIVisibilityCommands.GetLocalCommits).ConfigureAwait(false);
-        if (gitLogOutput is null)
-        {
-            Log.Warning("ITR: 'git log...' command is null");
-            return 0;
-        }
-
-        var localCommits = gitLogOutput.Output.Split(new[] { "\n" }, StringSplitOptions.RemoveEmptyEntries);
-        if (localCommits.Length == 0)
+        var commitsData = await GetCommitsAsync().ConfigureAwait(false);
```

Comment:
I know the spec shows to do all this again, but do we _need_ to fetch everything from the backend again, or could we reuse the remote commits to save making another async request?

Reply:
remote commits response depends on the local commits I send in the request payload, and because I'm unshallowing the local commits changes, so I'm forced to ask the backend again for the remote commits :/